backend/src/providers/network.ts

```diff
@@ -79,56 +80,53 @@ const mapValidators = (
 };
 
 const getEpochState = async (
-  currentEpochState: CurrentEpochState | null,
+  context: Context,
   epochStatus: RPC.EpochValidatorInfo,
   networkProtocolConfig: RPC.ProtocolConfigView
 ) => {
-  if (currentEpochState?.height === epochStatus.epoch_start_height) {
-    return currentEpochState;
+  if (
+    context.state.currentEpochState?.height === epochStatus.epoch_start_height
+  ) {
+    return context.state.currentEpochState;
   }
 
-  const { minimum_stake_ratio: epochMinStakeRatio } = await nearApi.sendJsonRpc(
-    "EXPERIMENTAL_genesis_config",
-    {}
-  );
   const maxNumberOfSeats =
     networkProtocolConfig.num_block_producer_seats +
     networkProtocolConfig.avg_hidden_validator_seats_per_shard.reduce(
       (sum, seat) => sum + seat,
       0
     );
-  currentEpochState = {
+  context.state.currentEpochState = {
     height: epochStatus.epoch_start_height,
-    seatPrice: nearApi.validators
-      .findSeatPrice(
-        epochStatus.current_validators,
-        maxNumberOfSeats,
-        epochMinStakeRatio,
-        networkProtocolConfig.protocol_version
-      )
-      .toString(),
+    seatPrice: context.state.genesis
+      ? nearApi.validators
+          .findSeatPrice(
+            epochStatus.current_validators,
+            maxNumberOfSeats,
+            context.state.genesis.minStakeRatio,
+            networkProtocolConfig.protocol_version
+          )
+          .toString()
+      : "0",
```

Comment:
This is the only downside of this PR.
In a quite unlikely case of a slow genesis block fetch we will have zeroed seatPrice for some time.

Reply:
Just checked - it is zero on staging.
Apparently not that unlikely 🤔

Reply:
That actually happened because I didn't try to recalculate seat price after genesis eventually resolved so I ended up with 0 seat price until next epoch.